### PR TITLE
feat(components/toast): tokenize toast styling

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**
[AB#3049569](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/3049569)

### CODE REVIEW

File: libs/components/indicators/src/lib/modules/alert/alert.modern.component.scss [hunk 1]
```diff
@@ -27,16 +27,16 @@
         --sky-override-alert-text-vertical-padding,
         var(--sky-space-inset-balanced-none)
       )
-      var(--sky-space-text_action_gap-m)
+      var(--sky-space-gap-text_action-m)
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/indicators/src/lib/modules/alert/alert.modern.component.scss:31]: These `gap` tokens changed in the most recent `design-tokens`.

File: libs/components/indicators/src/lib/modules/alert/alert.modern.component.scss [hunk 2]
```diff
@@ -27,16 +27,16 @@
...
         --sky-override-alert-text-vertical-padding,
         var(--sky-space-inset-balanced-none)
       )
-      var(--sky-space-icon_gap-l);
+      var(--sky-space-gap-icon-l);
     width: 100%;
 
     ::ng-deep a {
-      color: change-color($sky-text-color-default, $alpha: 0.8);
+      color: hsl(from var(--sky-text-color-default) h s l / 0.8);
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/indicators/src/lib/modules/alert/alert.modern.component.scss]: I discovered a different manner of doing this calculation which allows us to use the token. I did it in toast but also updated it here.

File: libs/components/indicators/src/lib/modules/alert/alert.modern.component.scss [hunk 3]
```diff
@@ -95,26 +95,26 @@
     border-style: solid;
     border-color: var(--sky-color-border-action-tertiary-base);
     border-radius: var(--sky-border-radius-s);
-    border-width: var(--sky-border-width-interactive-base);
+    border-width: var(--sky-border-width-action-base);
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/indicators/src/lib/modules/alert/alert.modern.component.scss:102]: These `border-width` tokens changed in the most recent `design-tokens`.

File: libs/components/theme/src/lib/styles/themes/modern/_elevation.scss [hunk 4]
```diff
@@ -36,8 +36,7 @@
   }
 
   .sky-elevation-24 {
-    box-shadow: $sky-theme-modern-elevation-24-shadow-size
-      $sky-theme-modern-elevation-24-shadow-color;
+    box-shadow: var(--sky-elevation-overlay-400);
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/theme/src/lib/styles/themes/modern/_elevation.scss:39]: We use this class in toast - updated here to allow us to tokenize when we did this component.

File: libs/components/theme/src/lib/styles/themes/modern/_theme.scss [hunk 5]
```diff
@@ -86,6 +86,6 @@
     }
   }
   .sky-rounded-corners {
-    border-radius: $sky-theme-modern-box-border-radius-default;
+    border-radius: var(--sky-border-radius-s);
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/theme/src/lib/styles/themes/modern/_theme.scss:89]: We use this class in toast - updated here to allow us to tokenize when we did this component.

File: libs/components/toast/src/lib/modules/toast/toast.component.scss [hunk 6]
```diff
@@ -106,47 +228,24 @@ sky-toast {
 
 .sky-theme-modern {
   sky-toast {
-    .sky-toast {
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/toast/src/lib/modules/toast/toast.component.scss:109]: I moved many of these styles up into the main declaration and used overrides when needed to clean up the code.

File: libs/components/toast/src/lib/modules/toast/toast.component.scss [hunk 7]
```diff
@@ -156,58 +255,11 @@ sky-toast {
 
     .sky-toast-icon-theme-modern {
       display: block;
-      margin: -5px 0 0 0;
-    }
-
-    .sky-box {
-      &.sky-toast-info {
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/toast/src/lib/modules/toast/toast.component.scss:163]: I updated our local function for these to allow for setting the icon color and then moved the pieces into the main declarations for these types to reduce code.

File: libs/components/toast/src/lib/modules/toast/toaster.component.scss [hunk 8]
```diff
@@ -1,29 +1,57 @@
...
+      )
+    );
   }
 
   @include mixins.sky-host-responsive-container-sm-min(false) {
     .sky-toaster {
-      width: 400px;
+      width: var(
+        --sky-override-toaster-width,
+        max(
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/toast/src/lib/modules/toast/toaster.component.scss:47]: Once we switch to V2 of modern we won't need to responsive container section here - but until then I had to specify the width calculation twice due to the override being different for the two sizes.

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2791/storybook/)

Component Storybooks:

- [a11y-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/a11y-storybook/)
- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/colorpicker-storybook/)
- [core-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/core-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/errors-storybook/)
- [flyout-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/flyout-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/forms-storybook/)
- [help-inline-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/help-inline-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/indicators-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/lookup-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/modals-storybook/)
- [navbar-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/navbar-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/phone-field-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/popovers-storybook/)
- [progress-indicator-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/progress-indicator-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/split-view-storybook/)
- [storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/theme-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/tiles-storybook/)
- [toast-storybook](https://blackbaud.github.io/skyux-pr-preview/2791/storybooks/toast-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2791/code-examples/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2791/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2791/playground/)

File: libs/components/toast/src/lib/modules/toast/toast.component.scss [hunk 9]
```diff
@@ -30,66 +133,94 @@ sky-toast {
...
+        var(--sky-color-text-default)
+      );
+      text-decoration: var(
+        --sky-override-toast-content-link-decoration,
+        var(--sky-font-text_decoration-visible_link)
+      );
 
       &:hover {
-        color: $sky-text-color-default;
+        color: var(--sky-text-color-default);
```
Comment (Blackbaud-ErikaMcVey) [inline libs/components/toast/src/lib/modules/toast/toast.component.scss]: I think this is one of the old tokens, and it should be `--sky-color-text-default` now

**Review by Blackbaud-ErikaMcVey — APPROVED** [review]

**Pull request merged**